### DISCUSSION TITLE
RustDesk service starting with gdm user and resulting in "Wrong password"

### DISCUSSION THREAD

**Discussion by livdelic** [rustdesk/rustdesk#9183] [General]
### Bug Description

I am encountering a critical issue with RustDesk where the service has started running under the gdm user instead of the usual admin user. Since this change, I am unable to log in, receiving a "Wrong password" error. This issue appeared recently, and it was working fine before when it was running under the admin user.

**Issue:** The RustDesk service is now starting under the gdm user account instead of the admin user. This change is causing login failures.
**Error Message:** "Wrong password" when attempting to log in to RustDesk.

before it was: 
`ps aux | grep rustdesk `
```
root         773  0.1  0.4 534784 34432 ?        Ssl  16:29   0:07 /usr/bin/rustdesk --service
root       55223  0.0  0.0  15152  6144 ?        S    17:29   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/128 -u gdm /usr/lib/rustdesk/rustdesk --server
gdm        55224  1.0  0.5 1954204 41688 ?       Sl   17:29   0:05 /usr/lib/rustdesk/rustdesk --server
root       59274  0.0  0.0   9564  2688 pts/1    S+   17:38   0:00 grep --color=auto rustdesk
```

now I get:
`ps aux | grep rustdesk `
```
root         791  0.1  0.4 534776 34560 ?        Ssl  16:27   0:07 /usr/bin/rustdesk --service
root       10868  0.0  0.0  15144  6144 ?        S    16:32   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/1000 -u adminuser /usr/lib/rustdesk/rustdesk --server
adminuser   10869  3.6  1.0 2249816 81068 ?       Sl   16:32   2:23 /usr/lib/rustdesk/rustdesk --server
adminuser   11090  0.3  2.9 3192924 229628 ?      Sl   16:33   0:12 /usr/lib/rustdesk/rustdesk --cm
root       92864  0.0  0.0   9564  2688 pts/1    S+   17:38   0:00 grep --color=auto rustdesk
```

**Connection from local to remote:** 

**Local:**
tried from win, ubuntu and mac

**Remote:**
Local:
OS = Ubuntu 24.04
Kernel = 6.8.0-39-generic
Display Server = X11 (with dummy display: https://github.com/rustdesk/rustdesk/issues/3592) 

### How to Reproduce

-

### Expected Behavior

Service User: The RustDesk service should start under the designated admin user account, not gdm (as far as I know).
Login Success: Users should be able to log in successfully with their credentials without receiving the "Wrong password" error.

### Operating system(s) on local side and remote side

Ubuntu 24.04

### RustDesk Version(s) on local side and remote side

1.2.6 & 1.3.0

### Screenshots

<img width="351" alt="image" src="https://github.com/user-attachments/assets/1486d3e5-9f1d-4228-a9a6-92aab9480523">

### Additional Context

Rustdesk server and client where not updated this weekend

service look like this: 
```
[Unit]
Description=Rustdesk
Requires=network.target
After=systemd-user-sessions.service

[Service]
Type=simple
ExecStart=/usr/bin/rustdesk --service
# kill --tray and --server both
ExecStop=pkill -f "rustdesk --"
# below two lines do not work, have to use above one line
#ExecStop=pkill -f "rustdesk --tray"
#ExecStop=pkill -f "rustdesk --server"
PIDFile=/run/rustdesk.pid
KillMode=mixed
TimeoutStopSec=30
User=root
LimitNOFILE=100000

[Install]
WantedBy=multi-user.target
```

I also tried the headless feature without success: https://github.com/rustdesk/rustdesk/pull/3902
service runs with the user lightdm & dummy display was of course disabled. 

Updated to 1.3.0 and still have the same problem.

**Comment by livdelic** [comment]
Problem solved: The Problem is solved. 
Problem was somehow the clients new used strategies, which they never did before. 

Change the password type to: "Use both passwords", restarted the service on the clients and it worked.

**Reply by GodsBest** [reply]
This worked for me. I had **Use permanent password** selected
This setup worked fine in v1.37 and earlier but stopped working in v1.38.
Selecting **Use both passwords** fixed the issue. However, I had to restart the target computer as restarting the rustdesk service did not seem to resolve the issue.
Thanks @livdelic

**Reply by 21pages** [reply]
Could you try out https://github.com/21pages/rustdesk/releases/tag/password and show the logs started with `====DEBUG====`? Nothing changed related to password between 1.3.6 and 1.3.8. 

https://github.com/rustdesk/rustdesk/wiki/FAQ#access-logs

https://github.com/21pages/rustdesk/commit/003d0181865fb2e6502e4cb13dd3d9d57945dbe3#diff-e9fac82dbc5f0a4bb05db748bd6958ed97d098730f84365f0b974020495c7d58

**Comment by 4piu** [comment]
I have a similar issue in `v1.4.7` where the rustdesk attempts to start server with `lightdm` user:
```
Jun 03 07:18:15 cvm systemd[1]: rustdesk.service: Consumed 4min 27.473s CPU time.
Jun 03 07:18:15 cvm systemd[1]: Started rustdesk.service - RustDesk.
Jun 03 07:18:24 cvm sudo[3497383]:     root : PWD=/ ; USER=root ; COMMAND=/usr/bin/env
Jun 03 07:18:24 cvm sudo[3497383]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jun 03 07:18:24 cvm sudo[3497383]: pam_unix(sudo:session): session closed for user root
Jun 03 07:18:24 cvm sudo[3497385]:     root : PWD=/ ; USER=lightdm ; ENV=XDG_RUNTIME_DIR=/run/user/113 ; COMMAND=/usr/share/rustdesk/rustdesk --server
Jun 03 07:18:24 cvm sudo[3497385]: pam_unix(sudo:session): session opened for user lightdm(uid=113) by (uid=0)
```
